rusk/src/lib/chain/rusk.rs

```diff
@@ -108,9 +114,18 @@ impl Rusk {
         let mut event_hasher = Sha3_256::new();
 
         for unspent_tx in txs {
+            if started.elapsed() > timeout {
+                info!("execute_transactions timeout triggered {timeout:?}");
+                break;
+            }
             // Don't include transactions if migration is triggered otherwise
             // the session rollback will not re-execute migration
             if Some(block_height) == self.migration_height {
+                info!("Skipping transactions due to migration_height");
+                break;
```

Comment:
When we accept this block we still delete its transactions from mempool.

```
 for tx in mrb.inner().txs().iter() {
                    let _ = Mempool::delete_tx(t, tx.hash())
 ```

Reply:
But there will be no transactions in that block, so it's safe.

The skip is only in the generate phase

Reply:
Indeed, the candidate block will have empty txs list.